### ISSUE TITLE
Can't activate SuperDuper mode while using Russian keyboard

### ISSUE THREAD

**Issue by aantipov** [jasonrudolph/keyboard#59]
## Problem

When switching to Russian language, I can't activate SuperDuper mode.
At the same time, WindowsLayout and Markdown modes work just fine.

I use macOS Catalina

Is there any way to make it working?

**Comment by jasonrudolph** [comment]
> When switching to Russian language...

👋 Hi @aantipov: Can you describe the specific steps for switching to Russian language? (I want to make sure I'm following the same exact steps you're following.) If you can provide some screenshots, that would be especially helpful.

**Comment by aantipov** [comment]
Hi @jasonrudolph 
He is a gif. Hope it'll help

![keyboard](https://user-images.githubusercontent.com/427621/68452375-931baf80-01f2-11ea-97c5-eaf5e99980c3.gif)

**Comment by jasonrudolph** [comment]
@aantipov: Thank you for sharing that gif. That's really helpful! 😅

I tried to reproduce the issue, but I'm unable to reproduce it. This is the behavior I'm seeing:

![out](https://user-images.githubusercontent.com/2988/68515983-bd03bf00-0250-11ea-8d16-482da667b8dc.gif)

Note: You mentioned that you're using macOS Catalina. I'm currently using macOS Mojave (10.14.6), so maybe the issue is specific to Catalina?

**Comment by aantipov** [comment]
@jasonrudolph Thanks for trying to help me. 
As per the gif you provided, you are not actually changing the current language, you are just highlighting the InputSource. The symbols typed in the "Input area" are always English.

Try to change the language using the Language icon in the topbar (as I did in my gif) or using a keyboard shortcut

**Comment by jasonrudolph** [comment]
> Try to change the language using the Language icon in the topbar (as I did in my gif) or using a keyboard shortcut

@aantipov: Thanks for that clarification. That allowed me to successfully reproduce the issue.

Currently, (S)uper (D)uper Mode looks for specific English characters. For example:

https://github.com/jasonrudolph/keyboard/blob/2d2b159f071f7a0461874defd4ba966974aa72a8/hammerspoon/super.lua#L35-L37

When using an English keyboard, this works. When using a non-English keyboard, it doesn't work. Thanks for identifying this issue! :bow:

---

If anyone has some time to investigate a fix, I think this could be resolved by updating [`super.lua`](https://github.com/jasonrudolph/keyboard/blob/2d2b159f071f7a0461874defd4ba966974aa72a8/hammerspoon/super.lua) to look at keycodes instead of characters. For example, the excerpt above would become something like this:

```lua
  local keyCode = event:getKeyCode()

  if keyCode == hs.keycodes.map['s'] then
```

You can see the related API docs at:

- https://www.hammerspoon.org/docs/hs.eventtap.event.html#getKeyCode
- https://www.hammerspoon.org/docs/hs.keycodes.html#map

There are a handful of places where `super.lua` currently uses `event:getCharacters()`, so all of those places will likely need to be updated to use keycodes instead.

**Comment by jasonrudolph** [comment]
This is likely resolved by #61, so I'm going to close this issue.

If you're still able to reproduce the problem using the latest code on the `master` branch, please let me know, and I'll re-open this issue. :bow:

**Comment by aantipov** [comment]
It works now with the latest update, so the problem was fixed. Thanks!
